### DISCUSSION TITLE
Cast to ObjectId failed error in GraphQL

### DISCUSSION THREAD

**Discussion by enricobuehler** [strapi/strapi#6269] [Legacy Topics]
Disclaimer: Starting a discussion because im not sure if this is a strapi related issue. 

Im trying to edit one of the core controllers of a review content-type. It has a relation field to the integrated user type, since it exposes all fields by default (including mail and everything else) i want to only respond with some public information. The controller works fine when using REST but when using GraphQL it repopulates the user field with the whole entry data. When removing the _id field of the user i get what seems to be a mongoose error, with `Cast to ObjectId failed for value \"[object Object]\" at path \"_id\" for model \"user\"`.  Also tried out disabling autoPopulate in the model settings, but still the same error.

Thanks in advance for any help, couldn't find anything regarding the exclusion of user fields in a response.

Steps to reproduce:

1. Create content type with user relation, edit content types controller (e.g. find) to remove user fields. 

2. Request the content type via GraphQL

Code Snippets:

The controller: 

```
async find(ctx) {
    let entities;
    if (ctx.query._q) {
      entities = await strapi.services.review.search(ctx.query, ["user"]);
    } else {
      entities = await strapi.services.review.find(ctx.query, ["user"]);
    }

    return entities.map(entity => {
      console.log(entity);

      let review = sanitizeEntity(entity, {
        model: strapi.models.review
      });
      
      if (review.user) {
        // Sanitizing the user entry by removing everything except public data.
        review.user = sanitizeReviewUser(review.user);
      }

      return review;
    });
  }

```

The sanitizeReviewUser function:

```
const sanitizeReviewUser = user => {

  const userData = JSON.parse(user.userData);

  const sanitizedUser = {
    username: user.username,
    userData: JSON.stringify({
      firstName: userData.firstName,
      lastName: userData.lastName.slice(0, 1) + "."
    })
  };

  return sanitizedUser;
};

```

Node version: v12.13.0
Strapi version: beta-20.3
NPM version: 6.12.0
Database: mongoDB 4.2.5
OS: Ubuntu 18.04.3 LTS
(Running in Docker)

Edit/Note: Changing the populate array also doesn't fix the problem since the problem occurs after calling the controller not while calling the search & find services

**Comment by lauriejim** [comment]
Can you send me you `api/review/models/review.settings.json` and `extensions/users-permissions/models/user.settings.json` files.
I don't really understand what you tried to do.
Can you try to add `"private": true` in your model for attributes you want to hide.
Maybe could work in your case.
